modules/javafx.graphics/src/main/java/com/sun/javafx/application/LauncherImpl.java

```diff
@@ -171,7 +171,7 @@ public static void launchApplication(final Class<? extends Application> appClass
             final Class<? extends Preloader> preloaderClass,
             final String[] args) {
 
-        if (launchCalled.getAndSet(true)) {
+        if (com.sun.glass.ui.Application.isEventThread() || launchCalled.getAndSet(true)) {
```

Comment:
Can you do this as two separate `if` checks, with the `isEventThread` check first? The error message for this case should be something like "Application launch must not be called on the JavaFX Application Thread".